src/System.Reflection.Metadata/src/System/Reflection/Metadata/PEReaderExtensions.cs

```diff
@@ -85,7 +85,7 @@ public static unsafe MetadataReader GetMetadataReader(this PEReader peReader, Me
             }
 
             var metadata = peReader.GetMetadata();
-            return new MetadataReader(metadata.Pointer, metadata.Length, options, utf8Decoder);
+            return new MetadataReader(metadata.Pointer, metadata.Length, options, utf8Decoder, memoryOwner: peReader);
```

Comment:
This remark is now out-of-date. Ditto for MetadataReaderProvider?

         /// The caller must keep the <see cref="PEReader"/> alive and undisposed throughout the lifetime of the metadata reader

Reply:
"undisposed" is still true, but can now leave it to finalizer safely. Right?

Reply:
Yes. Would still be flagged by IDisposable FxCop rule. I'll remove the "alive"